src/server.c

```diff
@@ -3210,11 +3210,11 @@ int processCommand(client *c) {
     }
 
     /* check if DEBUG/MODULE command not allowed */
-    if ((c->cmd->proc == debugCommand && !allowProtectedAction(server.enable_debug_cmd, c)) ||
-        (c->cmd->proc == moduleCommand && !allowProtectedAction(server.enable_module_cmd, c))) {
-        rejectCommandFormat(c,"%s command not allowed, you may enable it by "
-                              "editing the %s option in the Redis configuration file,"
-                              "and then restarting the server.",
+    if ((c->cmd->proc == debugCommand && c->cmd->flags & CMD_PROTECTED && !allowProtectedAction(server.enable_debug_cmd, c)) ||
+        (c->cmd->proc == moduleCommand && c->cmd->flags & CMD_PROTECTED && !allowProtectedAction(server.enable_module_cmd, c))) {
+        rejectCommandFormat(c,"%s command not allowed. If the %s option is set to \"local\","
+                              "you can run it from a local connection, otherwise you need to set this option"
+                              "in the configuration file, and then restart the server.",
```

Comment:
this is now a bit odd.
we have a global CMD_PROTECTED flag, but we have a per-command config flag (one for DEBUG and one for MODULE).

The fact is, that the only reason we added this flag is so that we can distinguish between different sub-commands on MODULE, without listing them all here, and i would still like that.
However, if we make that "protected" flag public (by exposing it in COMMAND output), it'll be a bit odd.
@guybe7 are we exposing all flags automatically? or do we have an explicit list of flags we expose?

anyway, in order to make that code look a bit nicer, we can add a first `if (c->cmd->flags & CMD_PROTECTED) {`
and put all the specific logic (about matching the cmd->proc and the right config var) inside it.
@YaacovHazan WDYT?

Reply:
we expose only the flags that appear in `addReplyFlagsForCommand`

Reply:
Agree that it's a bit odd but having a per-command flag is overkill. ill take your suggestion of rearrange the `if's`

Reply:
@guybe7 we need to go over that list and see if something is missing. let's look into it in the PR about the client hints. i'll add a todo